packages/plugins/src/hardhat/deployments.ts

```diff
@@ -200,30 +201,41 @@ export const deployChugSplashConfig = async (
 
   spinner.start('The deployment is being executed. This may take a moment.')
 
-  if (remoteExecution) {
-    await monitorTask(
+  // If executing locally, then startup executor with HRE provider and pass in canonical config
+  if (!remoteExecution) {
+    const executor = new ChugSplashExecutor()
+    executor.init()
+    executor.main(
       {
-        configPath,
-        silent: true,
+        privateKey:
+          '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80',
+        logLevel: silent ? 'info' : 'error',
       },
-      hre
+      provider,
+      canonicalConfig
     )
-  } else {
-    await executeTask(
-      {
-        chugSplashManager: ChugSplashManager,
-        bundleId,
-        bundle,
-        parsedConfig,
-        executor: signer,
-        silent: true,
-        isLocalExecution: true,
-      },
-      hre
+  }
+
+  // Monitor the deployment regardless
+  await monitorTask(
+    {
+      configPath,
+      silent: true,
+    },
+    hre
+  )
+
+  // If executing locally, complete post execution actions
+  if (!remoteExecution) {
+    const finalTxnHash = await getFinalDeploymentTxnHash(
+      ChugSplashManager,
+      bundleId
     )
-    await claimExecutorPayment(signer, ChugSplashManager)
+    await postExecutionActions(provider, parsedConfig)
+    await createDeploymentArtifacts(hre, parsedConfig, finalTxnHash)
```

Comment:
We already call `postExecutionActions` and `createDeploymentArtifacts` inside `monitorTask` above (on [these lines](https://github.com/chugsplash/chugsplash/blob/baa98515862f072791e7ee8e688697f96822cadc/packages/plugins/src/hardhat/tasks.ts#L869-L875)